src/FunctionApps/python/IntakePipeline/conversion.py

```diff
@@ -0,0 +1,153 @@
+        # a full message in next_message and need to append it
+        # to output. This will not trigger the first time we
+        # see a line with MSH since next_message will be empty
+        # at that time.
+        if next_message != "" and line.startswith("MSH"):
+            output.append(next_message)
+            next_message = ""
+
+        # Otherwise, continue to add the line of text to next_message
+        if line != "":
+            next_message += f"{line}\r"
+
+    # Since our loop only adds messages to output when it finds
+    # a line that starts with MSH, the last message would never
+    # be added. So we explicitly add it here.
+    if next_message != "":
+        output.append(next_message)
+
+    return output
+
+
+def convert_message_to_fhir(
+    message: str,
+    message_format: str,
+    message_type: str,
+    access_token: str
+) -> str:
+    message_format_map = {
+        "hl7v2": "Hl7v2",
+        "ccda": "Ccda",
+        "json": "Json"
+    }
+
+    message_type_map = {
+        "adt_a01": "ADT_A01",
+        "oml_o21": "OML_O21",
+        "oru_r01": "ORU_R01",
+        "vxu_v04": "VXU_V04",
+        "ccd": "CCD",
+        "ccda": "CCD",
+        "consultationnote": "ConsultationNote",
+        "dischargesummary": "DischargeSummary",
+        "historyandphysical": "HistoryandPhysical",
+        "operativenote": "OperativeNote",
+        "procedurenote": "ProcedureNote",
+        "progressnote": "ProgressNote",
+        "referralnote": "ReferralNote",
+        "transfersummary": "TransferSummary",
+        "examplepatient": "ExamplePatient",
+        "stu3chargeitem": "Stu3ChargeItem"
+    }
+
+    template_map = {
+        "Hl7v2": "microsofthealth/fhirconverter:default",
+        "Ccda": "microsofthealth/ccdatemplates:default",
+        "Json": "microsofthealth/jsontemplates:default",
+    }
+
+    input_data_type = message_format_map.get(message_format.lower())
+    root_template = message_type_map.get(message_type.lower())
+    template_collection = template_map.get(
+        input_data_type,
+        "microsofthealth/hl7v2templates:default"
+    )
+
+    if input_data_type is None or root_template is None:
+        raise Exception("Unknown file format or message type")
+
+    url = f"{get_required_config('FHIR_URL')}/$convert-data"
+    data = {
+        "resourceType": "Parameters",
+        "parameter": [
+            {"name": "inputData", "valueString": message},
+            {"name": "inputDataType", "valueString": input_data_type},
+            {"name": "templateCollectionReference", "valueString": template_collection},
+            {"name": "rootTemplate", "valueString": root_template}
+        ]
+    }
+    
+    data = json.dumps(data)
+
+    response = requests.post(
+        url=url,
+        data=data,
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {access_token}"
+        }
+    )
```

Comment:
```suggestion
    response = requests.post(
        url=url,
        json=data,
        headers={
            "Authorization": f"Bearer {access_token}"
        }
    )
```

Reply:
Good call! This is much cleaner and has now been implemented.